src/Optimization/hiopFactAcceptor.hpp

```diff
@@ -0,0 +1,124 @@
+// CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+// CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS
+// OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED
+// AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+// (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE,
+// EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Additional BSD Notice
+// 1. This notice is required to be provided under our contract with the U.S. Department
+// of Energy (DOE). This work was produced at Lawrence Livermore National Laboratory under
+// Contract No. DE-AC52-07NA27344 with the DOE.
+// 2. Neither the United States Government nor Lawrence Livermore National Security, LLC
+// nor any of their employees, makes any warranty, express or implied, or assumes any
+// liability or responsibility for the accuracy, completeness, or usefulness of any
+// information, apparatus, product, or process disclosed, or represents that its use would
+// not infringe privately-owned rights.
+// 3. Also, reference herein to any specific commercial products, process, or services by
+// trade name, trademark, manufacturer or otherwise does not necessarily constitute or
+// imply its endorsement, recommendation, or favoring by the United States Government or
+// Lawrence Livermore National Security, LLC. The views and opinions of authors expressed
+// herein do not necessarily state or reflect those of the United States Government or
+// Lawrence Livermore National Security, LLC, and shall not be used for advertising or
+// product endorsement purposes.
+//
+
+/**
+  * @file hiopFactAcceptor.cpp
+  *
+  * @author Nai-Yuan Chiang <chiang7@lnnl.gov>, LNNL
+  * @author Cosmin G. Petra <petra@lnnl.gov>, LNNL
+  *
+  */
+
+#ifndef HIOP_FACT_ACCEPTOR
+#define HIOP_FACT_ACCEPTOR
+
+#include "hiopNlpFormulation.hpp"
+#include "hiopPDPerturbation.hpp"
+
+namespace hiop
+{
+
+class hiopFactAcceptor
+{
+public:
+  /** 
+   * Default constructor 
+   * Determine if a factorization is acceptable or not
+   */
+  hiopFactAcceptor(hiopPDPerturbation* p)
+  : perturb_calc_{p}
+  {}
+
+  virtual ~hiopFactAcceptor() 
+  {}
+  
+  /** 
+   * @brief method to check if current factorization is acceptable or/and if
+   * a re-factorization is reqired by increasing 'delta_wx'-'delta_cd'. 
+   * 
+   * Returns '1' if current factorization is rejected
+   * Returns '0' if current factorization is ok
+   * Returns '-1' if current factorization failed due to singularity
+   */
+  virtual int requireReFactorization(const hiopNlpFormulation& nlp, const int& n_neg_eig,
+                                     double& delta_wx, double& delta_wd, double& delta_cc, double& delta_cd) = 0;
+  
+  inline void set_mu(const double& mu)
+  {
+    perturb_calc_->set_mu(mu);
+  }
+      
```

Comment:
I'm a little confused by our naming scheme here - should we stick with camelCase?

Reply:
I use camelCase for the new function `requireReFactorization` I created, but keep the one `set_mu` from its original design.
Anyhow, I deleted the 2nd method since I noticed that `set_mu` is not required anymore.